### ISSUE TITLE
Changing options of Select widget triggers Callback

### ISSUE THREAD

**Issue by nickpollari** [jupyter-widgets/ipywidgets#2230]
@jasongrout we should have a way to change the options of a Select widget without triggering the callback function. Whenever the options are changed for the widget the value attribute is automatically set which forces the callback to be triggered. See code below;

```python
from ipywidgets import Select

# callback
def cb(change):
    print(change)

# widget
s_w = Select(value=None, options=['a','b'])
s_w.observe(cb, names='value')

# triggers no callback
s_w.value = None
# triggers callback
s_w.options = ['1', '2', '3']
```

**Comment by crahan** [comment]
I'm currently running into this issue. I'm writing a file browser widget. As soon as I click a folder entry in the list it triggers the callback to set the options with the new folder contents as expected. However, the first option is always being selected and triggers a callback, making the code go into a loop as long as the automatically selected entry is a folder.

I'd be very interested in any potential workarounds until a more permanent solution is available.

**Comment by nickpollari** [comment]
@crahan my workaround has been to utilize the .unobserve method before updating the options.

```python
s_w.unobserve(cb, 'value')
s_w.options = ['1','2','3']
s_w.observe(cb, 'value')
```

**Comment by crahan** [comment]
@nickpollari that is awesome! Thank you so much. Everything's working as expected now. I added a `s_w.value = None` to make sure nothing is marked as selected.

```
dircontent.unobserve(on_dircontent_select, names='value')
dircontent.options = get_dir_contents(path,showhidden=False)
dircontent.observe(on_dircontent_select, names='value')
dircontent.value=None
```

**Comment by crahan** [comment]
Here's a screenshot of the final result.

![Screenshot 2019-04-04 at 21 41 39](https://user-images.githubusercontent.com/172588/55583605-8c8bfd80-5722-11e9-8fcf-951d2afa68d0.png)

**Comment by dreme** [comment]
And a +awesome vote from me too for the @nickpollari 'widget unobserve' solution. I was seeing the same problem as @crahan for a RadioButton widget. Every time I updated the widget options I would get _weird subtle bug behaviours_ due to the unexpected call-back trigger when the widget was updated.

**Comment by jasongrout** [comment]
Adding to 8.0 to implement the suggestion above at https://github.com/jupyter-widgets/ipywidgets/issues/2230#issuecomment-425693572

**Comment by nickpollari** [comment]
thanks @crahan @dreme and @jasongrout for the upcoming fix

**Comment by martinRenou** [comment]
Should fix #3284
